src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -515,6 +499,28 @@ async def build_pex(
     )
 
 
+def _validate_metadata(metadata: LockfileMetadata, request: PexRequest, python_setup: PythonSetup):
+    if metadata.is_valid_for(
+        request.requirements.lockfile_hex_digest,
+        request.interpreter_constraints,
+        python_setup.interpreter_universe,
+    ):
+        return None
+
+    message_1 = f"Invalid lockfile for PEX request `{request.output_filename}`. "
+    message_2 = (
```

Comment:
What's the motivation for two variables?

Reply:
`message_1` appears in the exception text, which I fell should generally be short enough to appear as the title to an error on sentry or such

Reply:
Oh, didn't see that. Imo, we can/should keep this all a single message. We have no precedent for something like using `logger.error()` + an exception. We always put all the info in the exception, which is often quite long and uses `\n\n` for it to be more readable.

Keep it simple here.

Reply:
I thought I _was_ keeping it simple :)

I